docker.go

```diff
@@ -177,23 +177,23 @@ func (p Plugin) Exec() error {
 	}
 
 	// add base image docker credentials to the existing config file, else create new
+	// instead of writing to config file directly, using docker's login func
 	if p.BaseImagePassword != "" {
-		json, err := setDockerAuth(p.Login.Username, p.Login.Password, p.Login.Registry,
-			p.BaseImageUsername, p.BaseImagePassword, p.BaseImageRegistry)
-		if err != nil {
-			return fmt.Errorf("Failed to set authentication in docker config %s", err)
-		}
-		os.MkdirAll(dockerHome, 0600)
-		path := filepath.Join(dockerHome, "config.json")
-		file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
-		if err != nil {
-			return fmt.Errorf("Error opening config.json: %s", err)
-		}
-		defer file.Close()
-		_, err = file.Write(json)
+		var baseConnectorLogin Login
+		baseConnectorLogin.Registry = p.BaseImageRegistry
+		baseConnectorLogin.Username = p.BaseImageUsername
+		baseConnectorLogin.Password = p.BaseImagePassword
+
+		cmd := commandLogin(baseConnectorLogin)
+
+		raw, err := cmd.CombinedOutput()
 		if err != nil {
-			return fmt.Errorf("Error writing config.json: %s", err)
+			out := string(raw)
+			out = strings.Replace(out, "WARNING! Using --password via the CLI is insecure. Use --password-stdin.", "", -1)
+			fmt.Println(out)
```

Comment:
What is this error message? If this is printed what's the change the customer has to do?

Reply:
just replacing the general docker warning, not an error.